src/nvim/eval.c

```diff
@@ -11537,18 +11474,18 @@ static void f_jobresize(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 
 
-  TerminalJobData *data = find_job(argvars[0].vval.v_number);
+  Channel *data = find_job(argvars[0].vval.v_number, true);
   if (!data) {
-    EMSG(_(e_invjob));
     return;
   }
 
-  if (data->proc.uv.process.type != kProcessTypePty) {
-    EMSG(_(e_jobnotpty));
+  if (data->type != kChannelTypeProc
```

Comment:
`find_job()` has already verified that `data->type == kChannelTypeProc`, so this condition could be removed.